### ISSUE TITLE
Why should the learning rate and weight decay affected by batchsize in this way?

### ISSUE THREAD

**Issue by sonack** [marvis/pytorch-yolo2#104]
Here 
https://github.com/marvis/pytorch-yolo2/blob/6c7c1561b42804f4d50d34e0df220c913711f064/train.py#L109

The lr is `learning_rate/batch_size`, while the weight_decay is `decay*batch_size`, I am confused by the this setting. I thought the larger batch size should enable the more accurate update direction, so enable larger lr, however it is opposite to what I thought. And I don't know why should set weight_decay as `decay*batch_size`. Can you explain this? Thanks for any rely!

**Comment by zHanami** [comment]
In my opinion, you will find the loss is sum but not the average for the batch, so it may get the same purpose when back prop. eg. w' = w - lr *dw(average) = w - lr/batch *dw(sum)
